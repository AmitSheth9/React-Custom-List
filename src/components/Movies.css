body {
    background: beige;
}

.container {
    border: 4px solid brown;
    background-color: yellow;
}
.movies {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    margin: 0;
    padding: 0;
    gap: 1rem;
    margin-top: 2rem;
}
.title {
    font-size: 20px;
    font-weight: 700;
}
img {
    width: 100%;
}