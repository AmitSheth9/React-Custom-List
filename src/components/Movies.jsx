import React from 'react'
import { Link } from 'react-router-dom'

export default function Movies({movies}) {
    return (
        <div>
            <ul>
            {movies.map((movie) => {
                return <li key={movie.id}>
                <Link to={`/movies/${movie.id}`}>  
                    <p>{movie.title}</p>
                    <img src={`https://image.tmdb.org/t/p/w500/${movie.backdrop_path}`} alt={movie.title}/> 
                </Link>
                </li>
            })}

            </ul> 
        </div>
    )
}
