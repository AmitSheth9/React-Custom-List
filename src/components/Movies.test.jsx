import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Movies from "./Movies";

const movies = [{
    title: 'Spider-Man: No Way Home',
    backdrop_path: 'https://image.tmdb.org/t/p/w500/1Rr5SrvHxMXHu5RjKpaMba8VTzi.jpg'
},
{
title: "Encanto",
backdrop_path: 'https://image.tmdb.org/t/p/w500/3G1Q5xF40HkUBJXxt2DQgQzKTp5.jpg'
}]

it('renders movies', () => {
    const container = render (
        <MemoryRouter initialEntries={['/']}>
            <Movies movies={movies}/>
        </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    
} );