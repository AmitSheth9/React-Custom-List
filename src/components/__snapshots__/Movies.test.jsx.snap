// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movies 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="movies"
        >
          <div
            class="container"
          >
            <a
              href="/movies/undefined"
            >
              <p
                class="title"
              >
                Spider-Man: No Way Home
              </p>
              <img
                alt="Spider-Man: No Way Home"
                src="https://image.tmdb.org/t/p/w500/https://image.tmdb.org/t/p/w500/1Rr5SrvHxMXHu5RjKpaMba8VTzi.jpg"
              />
            </a>
          </div>
          <div
            class="container"
          >
            <a
              href="/movies/undefined"
            >
              <p
                class="title"
              >
                Encanto
              </p>
              <img
                alt="Encanto"
                src="https://image.tmdb.org/t/p/w500/https://image.tmdb.org/t/p/w500/3G1Q5xF40HkUBJXxt2DQgQzKTp5.jpg"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="movies"
      >
        <div
          class="container"
        >
          <a
            href="/movies/undefined"
          >
            <p
              class="title"
            >
              Spider-Man: No Way Home
            </p>
            <img
              alt="Spider-Man: No Way Home"
              src="https://image.tmdb.org/t/p/w500/https://image.tmdb.org/t/p/w500/1Rr5SrvHxMXHu5RjKpaMba8VTzi.jpg"
            />
          </a>
        </div>
        <div
          class="container"
        >
          <a
            href="/movies/undefined"
          >
            <p
              class="title"
            >
              Encanto
            </p>
            <img
              alt="Encanto"
              src="https://image.tmdb.org/t/p/w500/https://image.tmdb.org/t/p/w500/3G1Q5xF40HkUBJXxt2DQgQzKTp5.jpg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
